### PR TITLE
[Fix] Magnetic callout for last point on LineChart

### LINKED ISSUE #29388: [Feature]: [LineChart] having a way to customize the "magnetism" arround the highlighted point
### Library

React / v8 (@fluentui/react)

### Describe the feature that you would like added

On the line chart, today it’s fairly hard to highlight the value of the last point, you need to be perfectly on it.

For example:

image

Selecting the last point :
image
and if you are near by, it’s the previous point that is highlighted:
image

Feature request: introduce some kind of “magnetism” props so when the pointer is “near” the point and in the magnetism area, it will highlight the point?
Happy to discuss further about it

### Expected output:
Add an invisible circle of 8px radius around the last point on line chart so that it is easier to see the hover card for the last point.
Need to take care of the case where the second last or last N points lie within the 8px circle range in which case do not show the circle of hover.

### Have you discussed this feature with our team

_No response_

### Additional context

_No response_

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Priority

Normal

### PR DESCRIPTION
**Pull request by Shubhabrata08**
## Previous Behavior

Screenshot 2023-10-09 162543

## New Behavior

Screenshot 2023-10-09 162407

Added a dummy large circle over last point to initiate clinging behaviour when mouse pointer is cose to last point.
## Related Issue(s)

- Fixes #29388

### CODE REVIEW

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 1]
```diff
@@ -792,42 +792,81 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
...
+                  opacity={isLegendSelected && !lastPointHidden ? 1 : 0.01}
+                  fill={this._getPointFill(lineColor, lastCircleId, j, true)}
+                  stroke={lineColor}
+                  strokeWidth={strokeWidth}
+                  role="img"
+                  aria-label={this._getAriaLabel(i, j)}
+                />
+                {/* Dummy circle acting as magnetic latch for last callout point */}
+                <circle
+                  id={lastCircleId}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/LineChart/LineChart.base.tsx:834]: create a new id for this circle. There should be no duplicate ids.

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 2]
```diff
@@ -792,42 +792,81 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
...
+                    x2,
+                    y2,
+                    verticaLineHeight,
+                    lastCirlceXCallout,
+                    lastCircleId,
+                    lastCirlceXCalloutAccessibilityData,
+                  )}
+                  onMouseOut={this._handleMouseOut}
+                  strokeWidth={0}
+                  role="img"
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/LineChart/LineChart.base.tsx:861]: role should be set as none

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx (hunk 1 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx:834]: Done

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx (hunk 2 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx]: Done

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 3]
```diff
@@ -792,42 +793,81 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
...
-                onBlur={this._handleMouseOut}
-                {...this._getClickHandler(this._points[i].data[j].onDataPointClick)}
-                opacity={isLegendSelected && !lastPointHidden ? 1 : 0.01}
-                fill={this._getPointFill(lineColor, lastCircleId, j, true)}
-                stroke={lineColor}
-                strokeWidth={strokeWidth}
-                role="img"
-                aria-label={this._getAriaLabel(i, j)}
-              />,
+              <React.Fragment key={lastCircleId + dummyCircleId}>
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/LineChart/LineChart.base.tsx:796]: Call it lastCircleId_container

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 4]
```diff
@@ -792,42 +793,81 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
...
+                  )}
+                  onMouseOut={this._handleMouseOut}
+                  strokeWidth={0}
+                  role="none"
+                  ref={(e: SVGCircleElement | null) => {
+                    this._refCallback(e!, circleId);
+                  }}
+                  focusable={false}
+                  onBlur={this._handleMouseOut}
+                  {...this._getClickHandler(this._points[i].data[0].onDataPointClick)}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/LineChart/LineChart.base.tsx:868]: we don't need click handler for the circle

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 5]
```diff
@@ -792,42 +793,81 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
...
+                    y2,
+                    verticaLineHeight,
+                    lastCirlceXCallout,
+                    lastCircleId,
+                    lastCirlceXCalloutAccessibilityData,
+                  )}
+                  onMouseOut={this._handleMouseOut}
+                  strokeWidth={0}
+                  role="none"
+                  ref={(e: SVGCircleElement | null) => {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/LineChart/LineChart.base.tsx:863]: why is ref needed

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx (hunk 3 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx:796]: As in create a separate key value for the fragment named lastCircleIdContainer?

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx (hunk 4 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx:868]: Ok... Will remove this...

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx (hunk 3 above)
Comment (AtishayMsft) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx:796]: {lastCircleId}_container

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx (hunk 5 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx:863]: Removing this

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 6]
```diff
@@ -792,42 +793,81 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
...
+                  stroke={lineColor}
+                  strokeWidth={strokeWidth}
+                  role="img"
+                  aria-label={this._getAriaLabel(i, j)}
+                />
+                {/* Dummy circle acting as magnetic latch for last callout point */}
+                <circle
+                  id={dummyCircleId}
+                  key={dummyCircleId}
+                  r={25}
```
[the remark this answers is not in the document]
Comment (Shubhabrata08) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx:837]: Done

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx (hunk 3 above)
Comment (Shubhabrata08) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx:796]: Done

File: packages/react-charting/src/components/LineChart/LineChart.base.tsx [hunk 7]
```diff
@@ -778,6 +778,7 @@ export class LineChartBase extends React.Component<ILineChartProps, ILineChartSt
           if (j + 1 === this._points[i].data.length) {
             // If this is last point of the line segment.
             const lastCircleId = `${circleId}${j}L`;
+            const dummyCircleId = `${circleId}${j}D`;
```
[the remark this answers is not in the document]
Comment (Shubhabrata08) [reply packages/react-charting/src/components/LineChart/LineChart.base.tsx:781]: Done

**Review by AtishayMsft — APPROVED** [review]

**Review by AtishayMsft — CHANGES_REQUESTED** [review]
Can we add a transparent borde to the existing circle and make it bigger to capture the nearby hover actions. This could eliminate the need for a new circle serving the same purpose.

**Comment by Shubhabrata08** [comment]
Ok... Will try implementing this

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**